# Resultierende Implementierung

## Hier finden die Imports statt.

In [18]:
import math
import numpy as np
import random as rd
import pandas as pd
from scipy import stats 

import matplotlib.pyplot as plt
import matplotlib as mat
import holoviews as hv

import vaex as vaex
import vaex.jupyter.model as vjm

import seaborn as sns

import time

## Die Daten werden hier eingelesen.
###### Mit Übergabe als <class 'vaex.dataframe.DataFrameArrays'> mit Ausgabe der dafür benötigten Zeit

Hier wurde die Datei zwar über Pandas importiert, jedoch in Vaex übernommen, da die Vaex eigene Importierung zeitlich mehr benötigt und noch eine größere Nachbereitung als der Import von Pandas bräuchte.

### annotation_example1.txt

In [19]:
start_proc = time.process_time()

datei_1 = pd.read_csv(
    '/Users/samel/Documents/Hochschule/aktuelles Semester/03. Projekt zur Datenanalyse/Datein/annotation_example1.txt',
    sep=";",
    header=0
    )

datei_1 = datei_1.rename(
    columns={'back_value                           ': 'backvalue'}
    )

df_1 = vaex.from_pandas(datei_1)
#print(type(daten))

ende_proc = time.process_time()
print('Benötigte Systemzeit: {:5.3f}s'.format(ende_proc-start_proc))

df_1


Benötigte Systemzeit: 6.207s


#          pos_id           barcode        run_id    test_date           sign    value    backvalue
0          '1D99900010425'  '1D9990001AA'  1         '2011-08-02 10:08'  'n'     102.48   99.97
1          '1D99900010426'  '1D9990001AA'  1         '2011-08-02 10:08'  'n'     89.8     106.2
2          '1D99900010427'  '1D9990001AA'  1         '2011-08-02 10:08'  'n'     91.11    104.31
3          '1D99900010428'  '1D9990001AA'  1         '2011-08-02 10:08'  'n'     86.9     107.12
4          '1D99900010429'  '1D9990001AA'  1         '2011-08-02 10:08'  'n'     88.18    100.59
...        ...              ...            ...       ...                 ...     ...      ...
3,985,915  '1DUMY04150201'  '1DUMY0415AE'  11        '2012-10-27 10:10'  'n'     90.98    104.34
3,985,916  '1DUMY04150202'  '1DUMY0415AE'  11        '2012-10-27 10:10'  'n'     103.84   110.09
3,985,917  '1DUMY04150203'  '1DUMY0415AE'  11        '2012-10-27 10:10'  'n'     109.63   100.68
3,985,918  '1DUMY04150204'  '1DUMY0415AE'  11        '2012-10-27 10:10'  'n'     100.13   101.13
3,985,919  '1DUMY04150205'  '1DUMY0415AE'  11        '2012-10-27 10:10'  'n'     125.69   86.37

### annotation_example2 (ursprünglich: annotation_example2-export-2020-04-30)

In [20]:
start_proc = time.process_time()

datei_2 = pd.read_csv(
    '/Users/samel/Documents/Hochschule/aktuelles Semester/03. Projekt zur Datenanalyse/Datein/annotation_example2.txt',
    sep=";",
    header=0
    )

datei_2 = datei_2.rename(
    columns={'back_value                           ': 'backvalue'}
    )

df_2 = vaex.from_pandas(datei_2)
#print(type(daten))

ende_proc = time.process_time()
print('Benötigte Systemzeit: {:5.3f}s'.format(ende_proc-start_proc))

df_2


Benötigte Systemzeit: 6.457s


#          pos_id         barcode        run_id    test_date           sign    value    backvalue
0          1BSP201640728  '1BSP24164HL'  1         '2013-02-01 06:02'  's'     65.05    50.57
1          1BSP201640729  '1BSP24164HL'  1         '2013-02-01 06:02'  's'     80.39    120.1
2          1BSP201640730  '1BSP24164HL'  1         '2013-02-01 06:02'  's'     98.83    97.27
3          1BSP201640731  '1BSP24164HL'  1         '2013-02-01 06:02'  's'     119.92   124.45
4          1BSP201640732  '1BSP24164HL'  1         '2013-02-01 06:02'  's'     94.73    81.88
...        ...            ...            ...       ...                 ...     ...      ...
4,073,467  1BSP800061435  '1BSP85006HH'  98        '2013-02-05 11:02'  's'     108.0    86.16
4,073,468  1BSP800061436  '1BSP85006HH'  98        '2013-02-05 11:02'  's'     99.36    80.8
4,073,469  1BSP800061437  '1BSP85006HH'  98        '2013-02-05 11:02'  's'     78.83    58.85
4,073,470  1BSP800061438  '1BSP85006HH'  98        '2013-02-05 11:02'  's'     122.61   104.87
4,073,471  1BSP800061439  '1BSP85006HH'  98        '2013-02-05 11:02'  's'     307.19   102.95

### annotation_example3.txt

In [21]:
start_proc = time.process_time()

datei_3 = pd.read_csv(
    '/Users/samel/Documents/Hochschule/aktuelles Semester/03. Projekt zur Datenanalyse/Datein/annotation_example3.txt',
    sep=";",
    header=0
    )

datei_3 = datei_3.rename(
    columns={'back_value                           ': 'backvalue'}
    )

df_3 = vaex.from_pandas(datei_3)
#print(type(daten))

ende_proc = time.process_time()
print('Benötigte Systemzeit: {:5.3f}s'.format(ende_proc-start_proc))

df_3


Benötigte Systemzeit: 8.435s


#          pos_id           barcode        run_id    test_date           sign    value    backvalue
0          '1001200171739'  '100129017HD'  10        '2017-07-07 03:07'  's'     88.33    131.05
1          '1001200171741'  '100129017HD'  10        '2017-07-07 03:07'  's'     98.66    95.69
2          '1001200171742'  '100129017HD'  10        '2017-07-07 03:07'  's'     35.03    70.9
3          '1001200171743'  '100129017HD'  10        '2017-07-07 03:07'  's'     99.09    79.79
4          '1001200171744'  '100129017HD'  10        '2017-07-07 03:07'  's'     82.09    143.5
...        ...              ...            ...       ...                 ...     ...      ...
5,314,555  '1002000172921'  '100209017HD'  9         '2017-07-07 09:07'  's'     88.9     128.91
5,314,556  '1002401401231'  '100249140HD'  9         '2017-07-07 09:07'  's'     93.52    91.56
5,314,557  '1002401402813'  '100249140HD'  9         '2017-07-07 09:07'  's'     88.53    93.61
5,314,558  '1BCS600562415'  '1BCS6Y056LD'  16        '2017-08-23 09:08'  's'     89.24    144.05
5,314,559  '1002001101220'  '100209110HD'  12        '2017-07-13 05:07'  's'     97.29    73.87

## Hier sollen die Dateien in Kategorien aufgeteilt werden
##### Nebenbei werden die Dateien mit den Variablen in 'df_zahl_ohne_f' übergeben, die die Datei ohne die Werte der Kategorie 'f' besteht und verwendet werden soll.

annotation_example1.txt:

In [22]:
f_1 = df_1[df_1.sign == 'f'] # f => failure (ein Problem mit einem der Geräte)

n_1 = df_1[df_1.sign == 'n'] # n => neutral (nur Lösungsmittel, keine Substanz)

p_1 = df_1[df_1.sign == 'p'] # p => positiv (kontrolle mit -max- Effekt, bzw. verschiede Konzentrationen -> DWK)

z_1 = df_1[df_1.sign == 'z'] # z => zero (negative kontrolle, oder zweite kontrolle, ditto)

s_1 = df_1[df_1.sign == 's'] # s => substanz (getestetes kleines chem. Molekül)


annotation_example2.txt:

In [28]:
f_2 = df_2[df_2.sign == 'f'] # f => failure (ein Problem mit einem der Geräte)

n_2 = df_2[df_2.sign == 'n'] # n => neutral (nur Lösungsmittel, keine Substanz)

p_2 = df_2[df_2.sign == 'p'] # p => positiv (kontrolle mit -max- Effekt, bzw. verschiede Konzentrationen -> DWK)

z_2 = df_2[df_2.sign == 'z'] # z => zero (negative kontrolle, oder zweite kontrolle, ditto)

s_2 = df_2[df_2.sign == 's'] # s => substanz (getestetes kleines chem. Molekül)


annotation_example3.txt:

In [24]:
f_3 = df_3[df_3.sign == 'f'] # f => failure (ein Problem mit einem der Geräte)

n_3 = df_3[df_3.sign == 'n'] # n => neutral (nur Lösungsmittel, keine Substanz)

p_3 = df_3[df_3.sign == 'p'] # p => positiv (kontrolle mit -max- Effekt, bzw. verschiede Konzentrationen -> DWK)

z_3 = df_3[df_3.sign == 'z'] # z => zero (negative kontrolle, oder zweite kontrolle, ditto)

s_3 = df_3[df_3.sign == 's'] # s => substanz (getestetes kleines chem. Molekül)


Hier kommt die Ausgabe der jeweiligen Längen:

In [29]:
# Der kommde Bereich ist für annotation_example1.txt ausgelegt
print('Länge des gesamten Datensatzes aus annotation_example1.txt:', len(df_1), '\n', '\n')
print('Länge der Kategorie f_1:', len(f_1), '\n')
print('Länge der Kategorie n_1:', len(n_1), '\n')
print('Länge der Kategorie p_1:', len(p_1), '\n')
print('Länge der Kategorie z_1:', len(z_1), '\n')
print('Länge der Kategorie s_1:', len(s_1), '\n')

print("Wobei wir die Kategorie 'f' vernachläsigen werden, da es sich hierbei um Gerätefehler handelt:")
print( len(df_1), '-' ,len(df_1[df_1.sign == 'f']), ' =', len(s_1) + len(z_1) + len(p_1) + len(n_1), '\n\n\n' )


# Der kommde Bereich ist für annotation_example2.txt ausgelegt
print('Länge des gesamten Datensatzes aus annotation_example2.txt:', len(df_2), '\n', '\n')
print('Länge der Kategorie f_2:', len(f_2), '\n')
print('Länge der Kategorie n_2:', len(n_2), '\n')
print('Länge der Kategorie p_2:', len(p_2), '\n')
print('Länge der Kategorie z_2:', len(z_2), '\n')
print('Länge der Kategorie s_2:', len(s_2), '\n')

print("Wobei wir die Kategorie 'f' vernachläsigen werden, da es sich hierbei um Gerätefehler handelt:")
print( len(df_2), '-' ,len(df_2[df_2.sign == 'f']), ' =', len(s_2) + len(z_2) + len(p_2) + len(n_2), '\n\n\n' )


# Der kommde Bereich ist für annotation_example3.txt ausgelegt
print('Länge des gesamten Datensatzes aus annotation_example3.txt:', len(df_3), '\n', '\n')
print('Länge der Kategorie f_3:', len(f_3), '\n')
print('Länge der Kategorie n_3:', len(n_3), '\n')
print('Länge der Kategorie p_3:', len(p_3), '\n')
print('Länge der Kategorie z_3:', len(z_3), '\n')
print('Länge der Kategorie s_3:', len(s_3), '\n')

print("Wobei wir die Kategorie 'f' vernachläsigen werden, da es sich hierbei um Gerätefehler handelt:")
print( len(df_3), '-' ,len(df_3[df_3.sign == 'f']), ' =', len(s_3) + len(z_3) + len(p_3) + len(n_3), '\n\n\n' )


Länge des gesamten Datensatzes aus annotation_example1.txt: 3985920 
 

Länge der Kategorie f_1: 638 

Länge der Kategorie n_1: 778292 

Länge der Kategorie p_1: 6144 

Länge der Kategorie z_1: 2368 

Länge der Kategorie s_1: 3198478 

Wobei wir die Kategorie 'f' vernachläsigen werden, da es sich hierbei um Gerätefehler handelt:
3985920 - 638  = 3985282 



Länge des gesamten Datensatzes aus annotation_example2.txt: 4073472 
 

Länge der Kategorie f_2: 36776 

Länge der Kategorie n_2: 663033 

Länge der Kategorie p_2: 8329 

Länge der Kategorie z_2: 0 

Länge der Kategorie s_2: 3365334 

Wobei wir die Kategorie 'f' vernachläsigen werden, da es sich hierbei um Gerätefehler handelt:
4073472 - 36776  = 4036696 



Länge des gesamten Datensatzes aus annotation_example3.txt: 5314560 
 

Länge der Kategorie f_3: 164522 

Länge der Kategorie n_3: 880350 

Länge der Kategorie p_3: 7744 

Länge der Kategorie z_3: 3872 

Länge der Kategorie s_3: 4258072 

Wobei wir die Kategorie 'f' vernachläsig

## Inteervalle festlegen, um zukünftig Ausreißer entfernen zu können.
 
### Intervalsgrenzen bestimmen
Die Intervalle beziehen sich auf die jeweilige Kategorie (n, p, z, s).

annotation_example1.txt:

In [33]:
print('Die Intervalle sind auf die jeweiligen signs bezogen und dem entsprechenden Feature, jedoch haben diese nicht den selben Intervall, was bei der Modelierung später angepasst werden muss:\n')

n_1_stdValue = n_1.std('value')
n_1_stdBackvalue = n_1.std('backvalue')
n_1_meanValue = n_1.mean('value')
n_1_meanBackvalue = n_1.mean('backvalue')

n_1_ValueGrenzen = ( n_1_meanValue - (3 * n_1_stdValue),
                    n_1_meanValue + (3 * n_1_stdValue) )

n_1_BackvalueGrenzen = ( n_1_meanBackvalue - (3 * n_1_stdBackvalue),
                        n_1_meanBackvalue + (3 * n_1_stdBackvalue) )

print('n:', '\n', 'VALUE:', n_1_ValueGrenzen, 'BACK_VALUE:', n_1_BackvalueGrenzen, '\n')



p_1_stdValue = p_1.std('value')
p_1_stdBackvalue = p_1.std('backvalue')
p_1_meanValue = p_1.mean('value')
p_1_meanBackvalue = p_1.mean('backvalue')

p_1_ValueGrenzen = ( p_1_meanValue - (3 * p_1_stdValue),
                    p_1_meanValue + (3 * p_1_stdValue) )

p_1_BackvalueGrenzen = ( p_1_meanBackvalue - (3 * p_1_stdBackvalue),
                        p_1_meanBackvalue + (3 * p_1_stdBackvalue) )

print('p:', '\n', 'VALUE:', p_1_ValueGrenzen, 'BACK_VALUE:', p_1_BackvalueGrenzen, '\n')



z_1_stdValue = z_1.std('value')
z_1_stdBackvalue = z_1.std('backvalue')
z_1_meanValue = z_1.mean('value')
z_1_meanBackvalue = z_1.mean('backvalue')

z_1_ValueGrenzen = ( z_1_meanValue - (3 * z_1_stdValue),
                    z_1_meanValue + (3 * z_1_stdValue) )

z_1_BackvalueGrenzen = ( z_1_meanBackvalue - (3 * z_1_stdBackvalue),
                        z_1_meanBackvalue + (3 * z_1_stdBackvalue) )

print('z:', '\n', 'VALUE:', z_1_ValueGrenzen, 'BACK_VALUE:', z_1_BackvalueGrenzen, '\n')



s_1_stdValue = s_1.std('value')
s_1_stdBackvalue = s_1.std('backvalue')
s_1_meanValue = s_1.mean('value')
s_1_meanBackvalue = s_1.mean('backvalue')

s_1_ValueGrenzen = ( s_1_meanValue - (3 * s_1_stdValue),
                    s_1_meanValue + (3 * s_1_stdValue) )

s_1_BackvalueGrenzen = ( s_1_meanBackvalue - (3 * s_1_stdBackvalue),
                        s_1_meanBackvalue + (3 * s_1_stdBackvalue) )

print('s:', '\n', 'VALUE:', s_1_ValueGrenzen, 'BACK_VALUE:', s_1_BackvalueGrenzen, '\n')


Die Intervalle sind auf die jeweiligen signs bezogen und dem entsprechenden Feature, jedoch haben diese nicht den selben Intervall, was bei der Modelierung später angepasst werden muss:

n: 
 VALUE: (64.76180857601881, 136.19746059287976) BACK_VALUE: (81.12422056855941, 119.28419146702588) 

p: 
 VALUE: (-56.69870487463294, 230.73459680171612) BACK_VALUE: (-73.99608706164176, 193.5839841970584) 

z: 
 VALUE: (-93.72026318241555, 193.71920743917232) BACK_VALUE: (-25.095635858411, 184.97798383138388) 

s: 
 VALUE: (51.25983086360952, 132.16165014704586) BACK_VALUE: (62.17253937809675, 114.33705804917648) 



annotation_example2.txt:

annotation_example3.txt:

In [32]:
print('Die Intervalle sind auf die jeweiligen signs bezogen und dem entsprechenden Feature, jedoch haben diese nicht den selben Intervall, was bei der Modelierung später angepasst werden muss:\n')

n_3_stdValue = n_3.std('value')
n_3_stdBackvalue = n_3.std('backvalue')
n_3_meanValue = n_3.mean('value')
n_3_meanBackvalue = n_3.mean('backvalue')

n_3_ValueGrenzen = ( n_3_meanValue - (3 * n_3_stdValue),
                    n_3_meanValue + (3 * n_3_stdValue) )

n_3_BackvalueGrenzen = ( n_3_meanBackvalue - (3 * n_3_stdBackvalue),
                        n_3_meanBackvalue + (3 * n_3_stdBackvalue) )

print('n:', '\n', 'VALUE:', n_3_ValueGrenzen, 'BACK_VALUE:', n_3_BackvalueGrenzen, '\n')



p_3_stdValue = p_3.std('value')
p_3_stdBackvalue = p_3.std('backvalue')
p_3_meanValue = p_3.mean('value')
p_3_meanBackvalue = p_3.mean('backvalue')

p_3_ValueGrenzen = ( p_3_meanValue - (3 * p_3_stdValue),
                    p_3_meanValue + (3 * p_3_stdValue) )

p_3_BackvalueGrenzen = ( p_3_meanBackvalue - (3 * p_3_stdBackvalue),
                        p_3_meanBackvalue + (3 * p_3_stdBackvalue) )

print('p:', '\n', 'VALUE:', p_3_ValueGrenzen, 'BACK_VALUE:', p_3_BackvalueGrenzen, '\n')



z_3_stdValue = z_3.std('value')
z_3_stdBackvalue = z_3.std('backvalue')
z_3_meanValue = z_3.mean('value')
z_3_meanBackvalue = z_3.mean('backvalue')

z_3_ValueGrenzen = ( z_3_meanValue - (3 * z_3_stdValue),
                    z_3_meanValue + (3 * z_3_stdValue) )

z_3_BackvalueGrenzen = ( z_3_meanBackvalue - (3 * z_3_stdBackvalue),
                        z_3_meanBackvalue + (3 * z_3_stdBackvalue) )

print('z:', '\n', 'VALUE:', z_3_ValueGrenzen, 'BACK_VALUE:', z_3_BackvalueGrenzen, '\n')



s_3_stdValue = s_3.std('value')
s_3_stdBackvalue = s_3.std('backvalue')
s_3_meanValue = s_3.mean('value')
s_3_meanBackvalue = s_3.mean('backvalue')

s_3_ValueGrenzen = ( s_3_meanValue - (3 * s_3_stdValue),
                    s_3_meanValue + (3 * s_3_stdValue) )

s_3_BackvalueGrenzen = ( s_3_meanBackvalue - (3 * s_3_stdBackvalue),
                        s_3_meanBackvalue + (3 * s_3_stdBackvalue) )

print('s:', '\n', 'VALUE:', s_3_ValueGrenzen, 'BACK_VALUE:', s_3_BackvalueGrenzen, '\n')


Die Intervalle sind auf die jeweiligen signs bezogen und dem entsprechenden Feature, jedoch haben diese nicht den selben Intervall, was bei der Modelierung später angepasst werden muss:

n: 
 VALUE: (67.79281497978246, 132.71558883688144) BACK_VALUE: (-86.29620736649795, 308.5234996934131) 

p: 
 VALUE: (-89.28200489630319, 191.30684993762543) BACK_VALUE: (-10235.07286609486, 16227.17839553701) 

z: 
 VALUE: (-86.97366500628111, 204.09270942776868) BACK_VALUE: (-236.47778804234548, 435.3835421745769) 

s: 
 VALUE: (34.632177276315836, 151.12981170836812) BACK_VALUE: (-410.9874897252928, 602.2923863216954) 



Das hier berechnete Intervall ist im Bezug auf die gesamte Datei ohne die Kategorie 'f', wobei sich hier nicht auf das jeweilige sign bezogen wird und somit problematisch sein könnte, aber zur Modllierung im späteren Verlauf besser zu verwenden ist und somit zukünftik damit gearbeitet wird.
Die vorherigen Intervallsgrenzen wurden zur Veranschaunlichung erstellt, um einen Vergleich bzw. Abweichung besser modelieren zu können.

In [7]:
df_ValueGrenzen = ( df.mean('value') - (3 * df.std('value')),
                   df.mean('value') + (3 * df.std('value')))

df_BackvalueGrenzen = ( df.mean('backvalue') - (3 * df.std('backvalue')),
                       df.mean('backvalue') + (3 * df.std('backvalue')) )

print('Wenn wir die Intevalle der importierten Datei bestimmen, was in df gespeichert wurde, kommt folgendes dabei raus:')
print('    df:', '\n    VALUE:', df_ValueGrenzen, 'BACK_VALUE:', df_BackvalueGrenzen, '\n')
print("    Mittelwert von 'value':", df.mean('value'), "\n    Mittelwert von 'backvalue':", df.mean('backvalue'), '\n\n')



df_ohne_f_ValueGrenzen = ( df_ohne_f.mean('value') - (3 * df_ohne_f.std('value')),
                          df_ohne_f.mean('value') + (3 * df_ohne_f.std('value')))

df_ohne_f_BackvalueGrenzen = ( df_ohne_f.mean('backvalue') - (3 * df_ohne_f.std('backvalue')),
                              df_ohne_f.mean('backvalue') + (3 * df_ohne_f.std('backvalue')) )

print('Wir verwenden vorhin beschrieben, jedoch die Intevalle der Datei ohne der Kategorie 'f', was in df_ohne_f gespeichert wurde und kommt folgendes dabei raus:')
print('    df_ohne_f:', '\n    VALUE:', df_ohne_f_ValueGrenzen, 'BACK_VALUE:', df_ohne_f_BackvalueGrenzen, '\n')
print("    Mittelwert von 'value':", df_ohne_f.mean('value'), "\n    Mittelwert von 'backvalue':", df_ohne_f.mean('backvalue'))


Wenn wir die Intevalle der importierten Datei bestimmen, was in df gespeichert wurde, kommt folgendes dabei raus:
    df: 
    VALUE: (38.07067313089432, 150.27277827053746) BACK_VALUE: (-713.5645279374443, 923.3391621498805) 

    Mittelwert von 'value': 94.17172570071568 
    Mittelwert von 'backvalue': 104.88731710621813 


Wir verwenden vorhin beschrieben, jedoch die Intevalle der Datei ohne der Kategorie , was in df_ohne_f gespeichert wurde und kommt folgendes dabei raus:
    df_ohne_f: 
    VALUE: (38.07129460869005, 150.03393152750283) BACK_VALUE: (-669.3395979545287, 874.6586361363945) 

    Mittelwert von 'value': 94.05261306809646 
    Mittelwert von 'backvalue': 102.65951909093293


=> Daraus ist erkenntlich, dass für den 'value' ein geringer Unterschied besteht ob der sign 'f' mit betrachtet wird oder nicht, wo hingegen es beim 'backvalue' es einen größeren Unterschied macht. Es werden zwar wenn wir die einzelnen sign's nochmal betrachten, einen Unterschied machen, ob wir sie zusammenhängend oder einzelnd betrachten, indem besipielsweise von den sign's n, z und s mehr Werte mit übernommen und betrachtet, wo es beim p so ist, dass es weniger werte sind, die betrachtet werden.

Um es genauer Auszudrücken: Das betrachten der Intervallsgrenzen der Datei ohne der Kateegorie 'f' sorgt dafür, dass wir 4995633 Zeilen von den gesamten 5314560 haben, wenn wir jedoch die Kategorien einzeln betrachtet und derren Zeilen zusammengefügt hätten, wären es stattdessen 4992498, was eine Abweichung von 3135 Zeilen ist.

In [9]:
df_cleaned_without_f = df_ohne_f[(df_ohne_f['value'] <= df_ohne_f_ValueGrenzen[1] )
                                 & (df_ohne_f['value'] >= df_ohne_f_ValueGrenzen[0] )
                                 & (df_ohne_f['backvalue'] <= df_ohne_f_BackvalueGrenzen[1] )
                                 & (df_ohne_f['backvalue'] >= df_ohne_f_BackvalueGrenzen[0] )
                                ]

print('  df_without_f:', len(df_ohne_f), '=> df_cleaned_without_f:', len(df_cleaned_without_f))#, '\n    -> Der Wert unterscheidet sich um', len(df_cleaned_without_f)-gesamt_anz, 'im verhältis zu der Menge, wo die signs einzeln betrachtet wurden.', '\n')



  df_without_f: 5150038 => df_cleaned_without_f: 4995633


In [1]:
%matplotlib inline
import numpy as np
from fcmeans import FCM
from matplotlib import pyplot as plt